index.js

```diff
@@ -37,6 +37,7 @@ function BinarySplit (matcher) {
       } else {
         if (offset >= buf.length) {
```

Comment:
Honestly, I don't see how this clause can ever be evaluated true (it doesn't in the tests and any scenario I can think of). So, the `offset = 0` two lines below is kind of a guess…